firmware/temperature_detection/temperature_detection.ino

```diff
@@ -132,13 +130,35 @@ int measure_distance(size_t samples, int interval) {
 float measure_temperature(size_t samples, int interval) {
     float measurements = 0;
     for(size_t counter = 0; counter < (samples || 1); counter++) {
-        measurements += temperature_sensor.readObjectTempC();
+        #if defined(SENSOR_MLX90614)
+            measurements += measure_temperature_mlx90614();
+        #elif defined(SENSOR_ZTP115M) && defined(ZTP115M_SENSOR_PIN)
+            measurements += measure_temperature_ztp115m();
+        #else
+            #error No sensor specified!
+        #endif
         delay(interval);
     }
     return measurements / (samples || 1); // celsius degrees
 }
 
 
+float measure_temperature_ztp115m() {
+    int sensorValue = analogRead(ZTP115M_SENSOR_PIN);
+    float voltage = ZTP115M_SENSOR_REFERENCE * (sensorValue / 4096.0);
+    return (15194.2 + (voltage + 106.093) * voltage) / 12055.99;
```

Comment:
I think there is something wrong with the formula here and the one before also, if we take as an example that the sensor output 1.36V the formula return a temperature of 1.27242. Maybe I'm missing something @jaihsingh ?

Reply:
I've trusted blindly the formula, but @jaihsingh might need to document it a bit.

Reply:
Update: after pulling some data from the datasheet, I realized that @jaihsingh's regression was flawless, but it did the conversion the other way around (temperature to voltage).

Reply:
The bad news is that the real error can be up to 6 ºC for sane polynomial degrees. We can either implement a look-up table or paste a ridiculously long formula and pray to the compiler to keep things optimized and sane.

Reply:
```suggestion
    return (
        - 3484.051933 * pow(voltage, 0)
        + 11296.70621 * pow(voltage, 1)
        - 15853.08557 * pow(voltage, 2)
        + 12490.72137 * pow(voltage, 3)
        - 5998.724365 * pow(voltage, 4)
        + 1797.550087 * pow(voltage, 5)
        - 328.5445919 * pow(voltage, 6)
        + 33.53566265 * pow(voltage, 7)
        - 1.465912126 * pow(voltage, 8)
    );
```

Reply:
@1technophile and @jaihsingh, can you please test it?

Reply:
Hello,

Seems coherent for me, even if the temperature is lower if you compare it to the MLX.